core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -364,25 +377,15 @@ private void contractNodes() {
             }
         }
 
-        // Preparation works only once so we can release temporary data.
-        // The preparation object itself has to be intact to create the algorithm.
-        close();
-
         dijkstraTime += nodeContractor.getDijkstraSeconds();
         periodTime += periodSW.getSeconds();
         lazyTime += lazySW.getSeconds();
         neighborTime += neighborSW.getSeconds();
-        logger.info("took:" + (int) allSW.stop().getSeconds()
-                + ", new shortcuts: " + Helper.nf(nodeContractor.getAddedShortcutsCount())
-                + ", " + prepareWeighting
-                + ", dijkstras:" + nodeContractor.getDijkstraCount()
-                + ", " + getTimesAsString()
-                + ", meanDegree:" + (long) meanDegree
-                + ", initSize:" + initSize
-                + ", periodic:" + periodicUpdatesPercentage
-                + ", lazy:" + lastNodesLazyUpdatePercentage
-                + ", neighbor:" + neighborUpdatePercentage
-                + ", " + Helper.getMemInfo());
+        logStats(counter, updateCounter);
+
+        // Preparation works only once so we can release temporary data.
+        // The preparation object itself has to be intact to create the algorithm.
+        close();
```

Comment:
Should we make close private now?

Reply:
I think that would be better yes. btw @Nakaner mentioned it would sometimes be useful to have some protected hooks, for example at cleanUp() in GraphHopper.java https://media.ccc.de/v/2018-5257-eisenbahnrouting_mit_graphhopper#t=739 (around 12min)

Reply:
Yes, but we should not do this in general. We should start as closed as possible as we otherwise provide a broader API that is harder to evolve. Also inheritance should be less used and replaced with composition. All this means we should still provide hooks, but well thought out and only if there is a 'proper' use case. E.g. the GraphHopper class with all the "hooks" is simple but not that good IMO and needs refactoring at some point.